src/internal/minikubetestenv/client_factory.go

```diff
@@ -0,0 +1,102 @@
+package minikubetestenv
+
+import (
+	"context"
+	"sync"
+	"testing"
+
+	"github.com/pachyderm/pachyderm/v2/src/client"
+	"github.com/pachyderm/pachyderm/v2/src/internal/require"
+	"github.com/pachyderm/pachyderm/v2/src/internal/testutil"
+	"golang.org/x/sync/semaphore"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	poolSize        = 3
+	namespacePrefix = "test-cluster-"
+)
+
+var (
+	clusterFactory *ClusterFactory
+	setup          sync.Once
+)
+
+type ClusterFactory struct {
+	managedClusters   map[string]*client.APIClient
+	availableClusters map[string]bool
```

Comment:
I think this could be a` map[string]struct{}`.  I don't see this assigned to false anywhere only deleted from.